### PR TITLE
Add `rust_sitrep()` function

### LINKED ISSUE #168: `rust_sitrep()` or something similar
Help users understand their `cargo` situation.
https://github.com/extendr/rextendr/issues/104

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
Closes #168

Add `rust_sitrep()` function to report on the state of rust infrastructure to the best of our ability. Rely on `cargo` and `rustup` to determine versions of the build system, host architecture, available targets, and default toolchain. If anything is missing, suggest using `rustup`.
`rust_sitrep()` does not make an attempt to gues if the appropriate target is installed.

``` r
rextendr::rust_sitrep()
#> Rust infrastructure sitrep:
#> ✔ "rustup": 1.25.1 (bb60b1e89 2022-07-12)
#> ✔ "cargo": 1.68.2 (6feb7c9cf 2023-03-26)
#> ℹ host: x86_64-pc-windows-msvc
#> ℹ toolchain: stable-x86_64-pc-windows-msvc
#> ℹ targets: i686-pc-windows-gnu, x86_64-pc-windows-gnu, and
#>   x86_64-pc-windows-msvc
```

<sup>Created on 2023-05-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Here we pretend nothing is installed

``` r
rextendr::rust_sitrep()
#> Rust infrastructure sitrep:
#> ✖ "rustup": not found
#> ✖ "cargo": not found
#> ✖ Cannot determine host, toolchain, and targets without "rustup"
#> ✖ "cargo" is required to build rextendr-powered packages
#> ℹ It is recommended to use "rustup" to manage "cargo" and "rustc"
#> ℹ See <https://rustup.rs/> for installation instructions
```

<sup>Created on 2023-05-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

**Comment by CGMossa** [comment]
Omg so far this thing looks clean.. 👍

File: R/rust_sitrep.R [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  result <- tryCatch(
+    processx::run(cmd, args, error_on_status = FALSE),
+    error = \(...) list(status = -1)
+  )
+  if (result[["status"]] != 0) {
+    NA_character_
+  } else {
+    result$stdout
+  }
+}
```
Comment (eitsupi) [inline R/rust_sitrep.R:71]: This is not a big issue, but I believe the following function added in #271 could be replaced by this function.
How about moving this function to utils.R and replacing `cargo_command_available`?

https://github.com/extendr/rextendr/blob/bff80289ee9e4384160dba8a09f1b3ca6ea449ba/R/utils.R#L36-L48

Comment (Ilia-Kosenkov) [reply R/rust_sitrep.R:71]: Yep, I am looking at it right now. I can do that

**Review by CGMossa — APPROVED** [review]
LGTM

**Pull request merged**